distributed/tests/test_utils_comm.py

```diff
@@ -63,10 +61,57 @@ async def test_gather_from_workers_permissive_flaky(c, s, a, b):
     x = await c.scatter({"x": 1}, workers=a.address)
 
     rpc = await BrokenConnectionPool()
-    data, missing, bad_workers = await gather_from_workers({"x": [a.address]}, rpc=rpc)
+    data, missing = await gather_from_workers(["x"], s.get_who_has, rpc=rpc)
 
-    assert missing == {"x": [a.address]}
-    assert bad_workers == [a.address]
+    assert data == {}
+    assert missing == {"x"}
+
+
+@gen_cluster(client=True, nthreads=[], config=NO_AMM)
+async def test_gather_from_workers_cancelled_error(c, s):
+    """Something somewhere in the networking stack raises CancelledError while
+    gather_from_workers is running
+
+    See Also
+    --------
+    test_worker.py::test_gather_dep_cancelled_error
+    test_worker.py::test_get_data_cancelled_error
+    https://github.com/dask/distributed/issues/8006
+    """
```

Comment:
XREF #8013

Reply:
How is this test related to the PR?
I'm rather conserned about this test. I don't want us to artificially cancel the tasks. The "network stack" should not be raising a CancelledError at all without a reason. Catching those can have unwanted and hard to debug consequences. This test